### PR TITLE
[KEYCLOAK-6621] - Removing unnecessary code to process scopes from typed resources

### PR DESCRIPTION
**Pull request by pedroigor**

### CODE REVIEW

File: server-spi-private/src/main/java/org/keycloak/authorization/policy/evaluation/DefaultPolicyEvaluator.java [hunk 1]
```diff
@@ -77,11 +77,6 @@ public void evaluate(ResourcePermission permission, EvaluationContext executionC
             if (resource.getType() != null) {
                 evaluatePolicies(() -> policyStore.findByResourceType(resource.getType(), resourceServer.getId()), consumer);
```
Comment (tchambard) [inline server-spi-private/src/main/java/org/keycloak/authorization/policy/evaluation/DefaultPolicyEvaluator.java]: @pedroigor 

We identified that this block is also responsible for performances issues when doing entitlement or evaluation when many resources have the same type

File: server-spi-private/src/main/java/org/keycloak/authorization/policy/evaluation/DefaultPolicyEvaluator.java [hunk 2]
```diff
@@ -77,11 +77,6 @@ public void evaluate(ResourcePermission permission, EvaluationContext executionC
...
             }
-
-            if (scopes.isEmpty() && !resource.getScopes().isEmpty()) {
-                scopes.removeAll(resource.getScopes());
-                evaluatePolicies(() -> policyStore.findByScopeIds(resource.getScopes().stream().map(Scope::getId).collect(Collectors.toList()), resourceServer.getId()), consumer);
-            }
         }
 
         if (!scopes.isEmpty()) {
```
Comment (tchambard) [inline server-spi-private/src/main/java/org/keycloak/authorization/policy/evaluation/DefaultPolicyEvaluator.java]: @pedroigor 

We identified that this block is also responsible for performances issues when doing entitlement or evaluation when many resources share the same scopes

File: server-spi-private/src/main/java/org/keycloak/authorization/policy/evaluation/DefaultPolicyEvaluator.java (hunk 1 above)
Comment (pedroigor) [reply server-spi-private/src/main/java/org/keycloak/authorization/policy/evaluation/DefaultPolicyEvaluator.java]: I've added a comment on https://issues.jboss.org/browse/KEYCLOAK-6621. in general, you should not have too many permissions for a single type. At least not that impact performance so badly as you are suggesting.

I think we can also cache this query. Currently it is hitting the database every time.

Comment (pedroigor) [reply server-spi-private/src/main/java/org/keycloak/authorization/policy/evaluation/DefaultPolicyEvaluator.java]: Actually, results from this method are being cached already.

Comment (tchambard) [reply server-spi-private/src/main/java/org/keycloak/authorization/policy/evaluation/DefaultPolicyEvaluator.java]: OK for this point. We will not use type anymore for our current usage

**Comment by pedroigor** [comment]
@tchambard, I think that with these changes you can get a better performance even when using the type property as you were doing.

If you can check, I appreciate.

**Review by mposolda — DISMISSED** [review]
I am approving without checking the code deeply. I assume @sguilhen did it :)

**Review by sguilhen — APPROVED** [review]
Looks good to me

**Review by mposolda — APPROVED** [review]

**Pull request merged**